datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -251,6 +251,19 @@ def http(self):
 
         return self._http
 
+    @property
+    def http_remapper(self):
+        """
+        This is for updating default values before the creation of ``self.http``.
+        Every access makes a copy to avoid affecting other instances.
+        """
+        if self.HTTP_CONFIG_REMAPPER is None:
+            self.HTTP_CONFIG_REMAPPER = {}
+        else:
+            self.HTTP_CONFIG_REMAPPER = copy.deepcopy(self.HTTP_CONFIG_REMAPPER)
+
+        return self.HTTP_CONFIG_REMAPPER
```

Comment:
I’m afraid this extra layer of abstraction might create more complexity.

I think the way it's done here is fine: https://github.com/DataDog/integrations-core/pull/5289